src/Pages/archivePage/archive.js

```diff
@@ -1,27 +1,44 @@
 import React from 'react';
 import './archive.css';
 import { useArchiveNote } from '../../context/archive-note-context';
+import { useToastContext } from '../../context/toastContext';
 
 
 export default function Archive() {
   const archiveNotes = useArchiveNote();
+  const setArchiveNotes = useArchiveNote();
   const { reStoreArchiveNote , deleteArchiveNote } = useArchiveNote();
   const allArchiveNotes = archiveNotes.archiveNotes;
+  const notify = useToastContext();
+
+  const changeBgColor = (e , archiveNote) => {
+    setArchiveNotes(prev => prev.filter( note => note._id === archiveNote._id ? {...note , bgColor: e.target.value} : {...note , bgColor: ''}))
+  }
+
+  const restoreNote = (note) => {
+    reStoreArchiveNote(note);
+    notify("Restore To Main Notes" , {type:'success'});
+  }
+
+  const deleteNote = (note) => {
+    deleteArchiveNote(note);
+    notify("Remove Archive Notes" , {type:'success'});
+  }
   return (
     <div className="notes-container">
       <input className='search-bar' type="search" name="search" id="search" placeholder='Search...' />
         <h3>Archive</h3>
         <div className="pinned-notes">
           {allArchiveNotes.length === 0 ? "No Archive Notes" : allArchiveNotes.map((archiveNote) => (
-             <div className="note" key={archiveNote._id}>
+             <div className="note" key={archiveNote._id} style={{backgroundColor : archiveNote.bgColor}}>
              <div className="title">{archiveNote.title}</div>
```

Comment:
Suggestion: You can move jsx for single note to separate component. This way it will be more readable.